packages/scm-extra/src/browser/history/scm-history-widget.tsx

```diff
@@ -15,73 +15,91 @@
+        readonly toRevision?: string;
+        readonly fromRevision?: string;
+    };
+    readonly uri?: string;
+    readonly maxCount?: number;
+}
+
+export namespace ScmCommitNode {
     // tslint:disable-next-line:no-any
-    export function is(node: any): node is GitCommitNode {
-        return !!node && 'commitSha' in node && 'commitMessage' in node && 'fileChangeNodes' in node;
+    export function is(node: any): node is ScmCommitNode {
+        return !!node && 'commitDetails' in node && 'expanded' in node && 'selected' in node;
     }
 }
 
-export type GitHistoryListNode = (GitCommitNode | GitFileChangeNode);
+export type ScmHistoryListNode = (ScmCommitNode | ScmFileChangeNode);
 
 @injectable()
-export class GitHistoryWidget extends GitNavigableListWidget<GitHistoryListNode> implements StatefulWidget {
-    protected options: Git.Options.Log;
+export class ScmHistoryWidget extends ScmNavigableListWidget<ScmHistoryListNode> implements StatefulWidget {
+    protected options: HistoryWidgetOptions;
     protected singleFileMode: boolean;
     private cancelIndicator: CancellationTokenSource;
-    protected listView: GitHistoryList | undefined;
+    protected listView: ScmHistoryList | undefined;
     protected hasMoreCommits: boolean;
     protected allowScrollToSelected: boolean;
 
     protected status: {
         state: 'loading',
     } | {
         state: 'ready',
-        commits: GitCommitNode[];
+        commits: ScmCommitNode[];
     } | {
         state: 'error',
         errorMessage: React.ReactNode
     };
 
+    protected readonly toDisposeOnRepositoryChange = new DisposableCollection();
+
+    protected historySupport: ScmHistorySupport | undefined;
+
     constructor(
+        @inject(ScmService) protected readonly scmService: ScmService,
         @inject(OpenerService) protected readonly openerService: OpenerService,
-        @inject(GitCommitDetailOpenHandler) protected readonly detailOpenHandler: GitCommitDetailOpenHandler,
         @inject(ApplicationShell) protected readonly shell: ApplicationShell,
         @inject(FileSystem) protected readonly fileSystem: FileSystem,
-        @inject(Git) protected readonly git: Git,
         @inject(ScmAvatarService) protected readonly avatarService: ScmAvatarService,
         @inject(WidgetManager) protected readonly widgetManager: WidgetManager,
-        @inject(GitDiffContribution) protected readonly diffContribution: GitDiffContribution) {
+    ) {
         super();
-        this.id = GIT_HISTORY_ID;
-        this.scrollContainer = 'git-history-list-container';
-        this.title.label = GIT_HISTORY_LABEL;
-        this.title.caption = GIT_HISTORY_LABEL;
+        this.id = SCM_HISTORY_ID;
+        this.scrollContainer = 'scm-history-list-container';
+        this.title.label = SCM_HISTORY_LABEL;
+        this.title.caption = SCM_HISTORY_LABEL;
         this.title.iconClass = 'fa git-history-tab-icon';
         this.title.closable = true;
         this.addClass('theia-git');
```

Comment:
Should not we get rid of `git` notion in the scm extra?

Reply:
I have made the changes to the css.  I've also added another sentence to the change log for the breaking css changes.